### PR TITLE
Feature/nms for multiple robot classes

### PR DESCRIPTION
**Pull request by phinik**
## Proposed changes
If multiple robot classes exist, one may wish to perform nms across all robot classes instead of per robot class.

## Necessary checks
- [ ] Update poetry package version [semantically](https://semver.org/)
- [ ] Test on your machine

### CODE REVIEW

**Comment by Flova** [comment]
The CI is now working in a basic sense. Rebase/Merge with master to enable it.

File: yoeo/detect.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
 def detect_directory(model_path, weights_path, img_path, classes, output_path,
                      batch_size=8, img_size=416, n_cpu=8, conf_thres=0.5, nms_thres=0.5,
-                     multi_robot=False, first_robot_id=0):
+                     robot_class_ids: Optional[List[int]] = None):
```
Comment (Flova) [inline yoeo/detect.py:31]: Optional and list are not imported

Comment (phinik) [reply yoeo/detect.py:31]: https://github.com/bit-bots/YOEO/blob/eb6bd857df945465641662cf7edd07dbe1763623/yoeo/detect.py#L15 ?

Comment (Flova) [reply yoeo/detect.py:31]: Weird, I am pretty sure GitHub showed me an linter error yesterday and the import line was missing. Maybe it showed only a subset of the commits idk..

**Pull request merged**